src/devices/tuya.ts

```diff
@@ -20407,4 +20407,40 @@ export const definitions: DefinitionWithExtend[] = [
             ],
         },
     },
+    {
+        fingerprint: tuya.fingerprint("TS0601", ["_TZE284_udaucpdi"]),
+        model: "TS0601_boiler_6",
+        vendor: "AVATTO",
+        description: "Smart boiler switch",
+        extend: [tuya.modernExtend.tuyaBase({dp: true, forceTimeUpdates: true})],
+        exposes: [
+            e.switch(),
+            e.enum("level", ea.STATE_SET, ["T0", "T30", "T60", "T90", "T120", "TON"]).withDescription("LEVEL"),
+            e.numeric("countdown", ea.STATE_SET).withValueMin(0).withValueMax(7200).withUnit("s").withDescription("countdown"),
+            e.numeric("cur_current", ea.STATE).withUnit("mA").withDescription("Current"),
```

Comment:
Use, `e.current(), e.voltage(), e.power()` here. Make sure to correct the current divisor (`tuya.valueConverterBasic.scale(0, 30000, 0, 30000)`) accordingly (`A` is expected not `mA`)